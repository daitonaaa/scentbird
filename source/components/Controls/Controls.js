import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './Controls.scss';

import {
  resetOpenId,
  getItemsList,
  uncheckChilds,
  checkedFirstChilds,
} from 'actions/items';


export class Controls extends Component {

  static propTypes = {
    resetOpenId: PropTypes.func.isRequired,
    getItemsList: PropTypes.func.isRequired,
    uncheckChilds: PropTypes.func.isRequired,
    checkedFirstChilds: PropTypes.func.isRequired,
  };

  handleResetList = () => {
    const {
      resetOpenId,
      getItemsList,
    } = this.props;

    resetOpenId();
    getItemsList();
  }

  render() {
    const {
      uncheckChilds,
      checkedFirstChilds,
    } = this.props;

    return (
      <div className="controls">
        <div
          className="controls__btn"
          onClick={checkedFirstChilds}
        >
          Выделить первые дочерние элементы
        </div>
        <div
          onClick={uncheckChilds}
          className="controls__btn"
        >
          Снять выделение со всех элементов
        </div>
        <div
          onClick={this.handleResetList}
          className="controls__btn reset"
        >
          Вернуться к изначальным данным
        </div>
      </div>
    );
  }
}


const mapDispatchToProps = {
  resetOpenId,
  getItemsList,
  checkedFirstChilds,
  uncheckChilds,
};

export default connect(
  ()=>({}),
  mapDispatchToProps
)(Controls);
