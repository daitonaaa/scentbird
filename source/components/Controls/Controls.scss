.controls {
    display: flex;
    justify-content: space-between;
    margin-bottom: 10px;

    & div {
      box-sizing: border-box;
      margin: 4%;
      padding: 10px;
      width: 28%;
      background: #f1f1f1;
      font-size: 13px;
      text-align: center;
      border: solid 1px #dcdcdc;
      cursor: pointer;
      transition: .2s;

      &:hover {
        background: #a18aaf;
        color: #fff;
        border: solid 1px #a18aaf;
      }
    }
  }
