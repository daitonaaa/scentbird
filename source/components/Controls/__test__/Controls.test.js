import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Controls } from 'components/Controls/Controls';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });


describe('> > > COMPONENT - Controls', () => {
  it('#create snapshot', () => {
    const output = shallow(
      <Controls
        resetOpenId={()=>{}}
        getItemsList={()=>{}}
        uncheckChilds={()=>{}}
        checkedFirstChilds={()=>{}}
      />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
