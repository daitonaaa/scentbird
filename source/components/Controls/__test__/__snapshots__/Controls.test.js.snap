// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`> > > COMPONENT - Controls #create snapshot 1`] = `
<div
  className="controls"
>
  <div
    className="controls__btn"
    onClick={[Function]}
  >
    Выделить первые дочерние элементы
  </div>
  <div
    className="controls__btn"
    onClick={[Function]}
  >
    Снять выделение со всех элементов
  </div>
  <div
    className="controls__btn reset"
    onClick={[Function]}
  >
    Вернуться к изначальным данным
  </div>
</div>
`;
